### PR TITLE
Crosschain swap

### PR DESCRIPTION
**Pull request by nguyenhoaidanh**
image

### CODE REVIEW

**Review by nhd98z — CHANGES_REQUESTED** [review]

File: src/components/Announcement/Popups/TransactionPopup.tsx [hunk 1]
```diff
@@ -137,7 +137,8 @@ const SUMMARY: { [type in TRANSACTION_TYPE]: SummaryFunction } = {
   [TRANSACTION_TYPE.UNWRAP_TOKEN]: summary2Token,
   [TRANSACTION_TYPE.APPROVE]: summaryApprove,
   [TRANSACTION_TYPE.SWAP]: summary2Token,
-  [TRANSACTION_TYPE.BRIDGE]: summaryBridge,
+  [TRANSACTION_TYPE.BRIDGE]: summaryCrossChain,
```
[the remark this answers is not in the document]
Comment (nguyenhoaidanh) [reply src/components/Announcement/Popups/TransactionPopup.tsx:140]: same noti.
bridge and cross-chain swap are both called `cross-chain`. summaryCrossChain would be fined, I think.

File: src/components/Announcement/index.tsx [hunk 2]
```diff
@@ -205,6 +205,7 @@ export default function AnnouncementComponent() {
       const hasNewMsg = data?.numberOfUnread !== numberOfUnread
       if (hasNewMsg) {
         resetUnread(RTK_QUERY_TAGS.GET_PRIVATE_ANN_BY_ID)
+        resetUnread(RTK_QUERY_TAGS.GET_TOTAL_UNREAD_PRIVATE_ANN)
```
[the remark this answers is not in the document]
Comment (nguyenhoaidanh) [reply src/components/Announcement/index.tsx:208]: there are 2 ways to invalidate cache rtk query
- 1 . use [invalidatesTags](https://redux-toolkit.js.org/rtk-query/usage/automated-refetching#:~:text=The-,invalidatesTags,-property%20on%20a)
- 2 . dispatch action to invalidate cache
the above is the second one.

File: src/components/swapv2/LimitOrder/helpers.ts [hunk 3]
```diff
@@ -26,12 +26,21 @@ function parseFraction(value: string, decimals = RESERVE_USD_DECIMALS) {
 }
 
 // 1.00010000 => 1.0001
-export const removeTrailingZero = (value: string) => parseFloat(value).toString()
+export const removeTrailingZero = (num: string) => {
```
[the remark this answers is not in the document]
Comment (nguyenhoaidanh) [reply src/components/swapv2/LimitOrder/helpers.ts:29]: good point, I forgot regex can do it better.
your regex don't pass the case: 10.00, 10. 
I will choose this one `str.replace(/^([\d,]+)$|^([\d,]+)\.0*$|^([\d,]+\.[0-9]*?)0*$/, "$1$2$3")`

File: vite.config.ts [hunk 4]
```diff
@@ -55,4 +55,7 @@ export default defineConfig({
       util: 'util',
     },
   },
+  server: {
+    port: 3000,
+  },
```
Comment (hungdoansy) [inline vite.config.ts:60]: By the way, please add `--open` to open the browser when starting (Ref: https://vitejs.dev/guide/cli.html#options)

File: src/components/swapv2/LimitOrder/helpers.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (nguyenhoaidanh) [reply src/components/swapv2/LimitOrder/helpers.ts:29]: how about this Number('1.234000000000000000000000000001') :v

[the remark this answers is not in the document]
Comment (hungdoansy) [reply src/components/swapv2/LimitOrder/helpers.ts:29]:
```js
Number('1.2340000000000001') // 1.2340000000000002 
```
Well done js =))

File: src/components/swapv2/LimitOrder/helpers.ts [hunk 5]
```diff
@@ -29,12 +29,7 @@ export function parseFraction(value: string, decimals = RESERVE_USD_DECIMALS) {
 export const removeTrailingZero = (num: string) => {
   if (num === undefined || num === null) return ''
   num = String(num)
-  if (!num.includes('.')) return num
-  while (num[num.length - 1] === '0' || num[num.length - 1] === '.') {
-    if (!num.includes('.')) return num
-    num = num.substring(0, num.length - 1) // rm last
-  }
-  return num
+  return num.replace(/^([\d,]+)$|^([\d,]+)\.0*$|^([\d,]+\.[0-9]*?)0*$/, '$1$2$3')
```
Comment (nhd98z) [inline src/components/swapv2/LimitOrder/helpers.ts:32]: It's better if you explain this regex.

**Review by nhd98z — DISMISSED** [review]

File: src/components/swapv2/LimitOrder/helpers.ts (hunk 5 above)
Comment (nguyenhoaidanh) [reply src/components/swapv2/LimitOrder/helpers.ts:35]: updated

**Review by nhd98z — APPROVED** [review]

**Pull request merged**